sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -220,10 +221,28 @@ case class AdaptiveSparkPlanExec(
   }
 
   private def getExecutionId: Option[Long] = {
-    // If the `QueryExecution` does not match the current execution ID, it means the execution ID
-    // belongs to another (parent) query, and we should not call update UI in this query.
     Option(context.session.sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY))
-      .map(_.toLong).filter(SQLExecution.getQueryExecution(_) eq context.qe)
+      .map(_.toLong)
+  }
+
+  private lazy val shouldUpdatePlan: Boolean = {
+    // If the `QueryExecution` does not match the current execution ID, it means the execution ID
+    // belongs to another (parent) query, and we should call update metrics instead of plan in
+    // this query. For example:
+    //
+    //          ...
+    //           |
+    //  AdaptiveSparkPlanExec (query execution 0, no execution id)
+    //           |
+    //  InMemoryTableScanExec
+    //           |
+    //          ...
+    //           |
+    //  AdaptiveSparkPlanExec (query execution 1, execution id 0)
```

Comment:
I feel it's hacky to have nested `AdaptiveSparkPlanExec`. Shall we replace `AdaptiveSparkPlanExec` with `CacheTableQueryStageExec` before caching a plan?

Reply:
They are tow query executions. How can the outer AQE work through IMR if we replace the nested AQE with `CacheTableQueryStageExec`